### PR TITLE
[refactor] #4323: telemetry features, tokio console, logs, and configuration

### LINKED ISSUE #4323: [discussion] Remove `tokio-console` feature from `iroha_config`
`iroha_config` crate has `tokio-console` feature, which enables `logger.tokio_console_address` configuration parameter.

It introduces some dev issues:

- If you run `cargo tests` without this feature enables, config snapshot tests will fail, and their feature-gating will be quite... ambiguous and cumbersome, I can imagine.
- It produces some cratches across workspace, like this:
  https://github.com/hyperledger/iroha/blob/964476722e2a219becaacdb3676ca058ec5748cd/logger/src/lib.rs#L85-L90

A benefit of having this feature is that... allowing to compile Iroha without knowing that this field even exists?

From the UI standpoint, I think, even in this case it would be better to have `logger.tokio_console_address` field in the config anyway, but if the actual feature is disabled, Iroha can issue a warning. This is more user-friendly, simpler in the code, and has minimal overhead.

### PR DESCRIPTION
**Pull request by 0x009922**
## Description

_This is an improved iteration of https://github.com/hyperledger/iroha/pull/4369_

- Remove `logger.tokio_console_address` configuration parameter entirely: this could be configured in a more refined way using default ENVs (see [`console_subscriber` docs](https://docs.rs/console-subscriber/0.2.0/console_subscriber/struct.Builder.html#method.with_default_env))
- Rename `[telemetry.dev]` config section to `[dev_telemetry]`
- Update `telemetry` and `dev-telemetry` features of `iroha`
- Update `tokio-console` feature of `iroha_logger`: now console seamlessly starts whenever this feature is enabled
- Improve logs & warnings
- Fix dev-telemetry file logs: separate each item with `\n`, and recursively create parent directories
- Improve config validation: handle file paths

### Linked issue

Closes #4323 

### Benefits

- Less config parameters
- More flexible way to configure tokio console
- Less feature mess (?)
- Better warnings
- Refactoring

### Checklist

- [x] Document how to use tokio console
- [x] Figure out what happened to `Cargo.toml`s
- [x] Check manually several cases, how Iroha behaves with `telemetry`/`dev-telemetry` features enabled depending on `telemetry` configuration (warnings, logs, file output)
- [x] Check manually Tokio Console (works, but requires `TRACE` log level)

### CODE REVIEW

File: Cargo.toml
Comment (mversic) [inline Cargo.toml:1]: this file got messed up

File: core/Cargo.toml [hunk 1]
```diff
@@ -18,15 +18,10 @@ categories.workspace = true
...
 
 # Support lightweight telemetry, including diagnostics
 telemetry = []
-# Support the included CLI
-cli = []
-# Support developer-specific telemetry.
-# Should not be enabled on production builds.
-dev-telemetry = ["telemetry", "iroha_telemetry/dev-telemetry"]
 # Support Prometheus metrics. See https://prometheus.io/.
 expensive-telemetry = ["iroha_telemetry/metric-instrumentation"]
```
Comment (mversic) [inline core/Cargo.toml:26]: I think this isn't used and can be removed

Comment (0x009922) [reply core/Cargo.toml:26]: There are many `cfg`s in `iroha_telemetry_derive`'s generated code relying on this feature. I wonder why it isn't used by `iroha` crate, and how Prometheus works without it...

Comment (0x009922) [reply core/Cargo.toml:26]: There is still a mess around telemetry features...

- In Torii, `telemetry` enables `/version` (why it is telemetry?), `/status` (again, why?), and `/metrics` (Prometheus data) endpoints;
- In Iroha itself, `telemetry` enables an WebSocket service which submits a telemetry events to some endpoint. `dev_telemetry` extends previous case to writing to a file, and also enables tokio console in `iroha_logger`.
- `Metrics` in Sumeragi in `iroha_core`, which are served as Prometheus metrics, are collected anyway, independent from enabled features
- `iroha_core` has `telemetry` feature, which doesn't enable anything else, and isn't used inside by the crate
- purpose of `iroha_telemetry/metric-instrumentation` isn't really clear

File: Cargo.toml
Comment (0x009922) [reply Cargo.toml:1]: Fixed

File: cli/src/lib.rs [hunk 2]
```diff
@@ -229,16 +230,16 @@ impl Iroha {
...
                 Some(state)
             }
             Err(TryReadSnapshotError::NotFound) => {
-                iroha_logger::info!("Didn't find a snapshot of state, creating an empty one");
+                iroha_logger::info!("Didn't find a WSV snapshot; creating empty WSV");
                 None
             }
             Err(error) => {
-                iroha_logger::warn!(%error, "Failed to load state from a snapshot, creating an empty one");
+                iroha_logger::warn!(%error, "Failed to load WSV from a snapshot; creating empty WSV");
```
Comment (Erigara) [inline cli/src/lib.rs:242]: `WSV` is renamed to just `State` now, so referring to `WSV` doesn't really make sense now

**Comment by Erigara** [comment]
Does telemetry used at all? I'm aware that QA and DevOps team using prometheus metrics.
But i'm not aware of using WS for telemetry, if it's not required we can make code simpler.

**Comment by 0x009922** [comment]
@Erigara, created a separate discussion for it: https://github.com/hyperledger/iroha/issues/4385

File: core/Cargo.toml (hunk 1 above)
Comment (0x009922) [reply core/Cargo.toml:26]: @mversic, should we address it now, or separately, or never?

Comment (mversic) [reply core/Cargo.toml:26]: let 's open a ticket

Comment (0x009922) [reply core/Cargo.toml:26]: Opened: #4386

File: cli/src/lib.rs (hunk 2 above)
Comment (0x009922) [reply cli/src/lib.rs]: Fixed.

**Review by mversic — APPROVED** [review]

**Review by Erigara — APPROVED** [review]

**Pull request merged**